### ISSUE TITLE
Does not work when nested

### ISSUE THREAD

**Issue by binarykitchen** [christianalfoni/formsy-react#111]
I have this weird case:

```cjsx
  render: ->
    <div>
      <Row>
        <Input
          hasFeedback
          required
          name  = {@props.passwordName}
          label = {t("Password")}
          value = {@state.password}
        >
          <InputPassword
            required
            zxcvbn    = {zxcvbn}
            className = {@props.className}
            onChange  = {@_handlePasswordChange}
          />
        </Input>
      </Row>
      <Row>
        <Input
          hasFeedback
          required
          name  = {@props.repeatPasswordName}
          label = {t("Repeat Password")}
          help  = {@_getRepeatPasswordHelp()}
          value = {@state.repeatPassword}
        >
          <InputPassword
            required
            zxcvbn    = {false}
            infoBar   = {false}
            className = {@props.className}
            onChange  = {@_handleRepeatPasswordChange}
          />
        </Input>
      </Row>
    </div>
```

where the Input is from `require('react-bootstrap')` and InputPassword is from `require('react-ux-password-field')` ...

Bootstrap's Input allows it to have more inputs nested. But that does not go well with formsy-react and is probably the cause of https://github.com/christianalfoni/formsy-react/issues/110

**Comment by christianalfoni** [comment]
Hi @binarykitchen ,

You have to provide a Formsy.Form element too:

```js
<Formsy.Form>
        <MyInput
          hasFeedback
          required
          name  = {@props.passwordName}
          label = {t("Password")}
          value = {@state.password}
        >
</Formsy.Form>
```

**Comment by binarykitchen** [comment]
I already do in a parent element.

**Comment by christianalfoni** [comment]
It has to be inside the same component :-)

In React 0.14 we get parent-child context which will fix this, but currently it is not possible.

**Comment by binarykitchen** [comment]
ah, that is very unfortunate :(

**Comment by christianalfoni** [comment]
You are the first person to point it out actually ;-)

We are using Formsy on a huge project with a lot of forms and have not met that issue. That said, it would be nice to have, and it will when React 0.14 is released.

**Comment by binarykitchen** [comment]
alright, i ll have to search for another react form validation library for now or go with the native `setCustomValidity()` ...

**Comment by christianalfoni** [comment]
Jup, the important thing is that you find something that works for you :-)

**Comment by christianalfoni** [comment]
Just to make sure there was no misunderstanding here :-) :

This works:
```js
var MyForm = React.createClass({
  render: function () {
    return (
      <Formsy.Form>
        <div>
          <MyInput/>
        </div>
        <div>
          <div>
             <MyInput/>
           </div>
        </div>
      </Formsy.Form>
    );
  }
});
```

But this does not until React 0.14:

```js
var MyFormContent = React.createClass({
  render: function () {
    return (
        <div>
          <MyInput/>
        </div>
        <div>
          <div>
             <MyInput/>
           </div>
        </div>
    );
  }
});

var MyForm = React.createClass({
  render: function () {
    <Formsy.Form>
      <MyFormContent/>
    </Formsy.Form>
  }
});
```

**Comment by anavarro9731** [comment]
Great news. I have just encountered the problem as well with the "multi-step/wizard" control I am building (each step as its own control). formsy is fantastic BTW really fills a much needed gap IMO

**Comment by ziflex** [comment]
Same here :(

**Comment by christianalfoni** [comment]
React 0.14 is now in BETA and new formsy version is right around the corner. Will release some test info very soon.
